File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestArithSeqLength = function(nums) {
    let n = nums.length
    let arr = Array.from({length: n}, ()=>new Map())
    let res=0
    for(let i=n-2;i>=0;i--){
        for(let j=n-1;j>i;j--){
            let diff = nums[i]-nums[j]
            let count = arr[j].has(diff) ? arr[j].get(diff)+1 : 2
            arr[i].set(diff, count)
            res=Math.max(res, count)
        }
    }
    // console.log(arr)
    return res
};

File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.py
class Solution:
    def longestArithSeqLength(self, nums: List[int]) -> int:
        n = len(nums)
        arr = [{} for i in range(n)]
        res = 0
        for i in range(n-2, -1, -1):
            for j in range(n-1, i, -1):
                diff = nums[i]-nums[j]
                count = arr[j][diff]+1 if diff in arr[j] else 2
                arr[i][diff] = count
                res = max(res, count)
            
        
        # print(arr)
        return res